### PR TITLE
<ul type> & <ol type>

### LINKED ISSUE #11615: <ul type> and <ol type> not supported by List nor ListProperties
## 📝 Provide detailed reproduction steps (if any)

While bringing the `<ol start>`  and `<ol reversed>`  functionality to Drupal core, we’re also looking at `<ol type>` and `<ul type>`, because as https://ckeditor.com/blog/ckeditor-5-v32.0.0-with-new-list-properties-support-for-the-script-tag-and-enhanced-mentions/ mentions, this is now supported too and has a magnificent UX. 

https://ckeditor.com/docs/ckeditor5/latest/api/module_list_listproperties-ListPropertiesConfig.html#member-styles says:
> When set, the list style feature will be enabled. It allows changing the list-style-type HTML attribute of the lists.

But:
1. There is no `list-style-type` HTML attribute. Only a `type` attribute.
2. There seems to be no way to configure CKEditor 5 to generate a type attribute, only `list-style-type`.
3. Searching the CKE5 codebase finds plenty of matches for the strings `<ul style="list-style-type` and `<ol style="list-style-type`, but zero for `<ul type` and `<ol type`

**Conclusion:**

Because `ListProperties`' support for list style types does _not_ work for `<ol type>` or `<ul type>`, despite that being in the HTML spec:

- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ol#attr-type → not deprecated
- https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#attr-type → deprecated _but supported_

→ No downcast support, but also, more worryingly still, no upcast support.

Therefore the only work-around is to use GHS to add support for these. But … _that also does not work_:

1. Load the following CKE5 plugins: `htmlSupport.GeneralHtmlSupport`, `list.List`, with the following `htmlSupport` config:
```
{"allow":[{"name":"ul","attributes":{"type":true},"classes":true,"styles":true}]}
```
2. Instantiate CKEditor 5 on a textarea containing markup like:
```
<ol type="A">
  <li>test</li>
</ol>
```

**I strongly suspect that `v32` caused this to not work with GHS anymore, because I do recall testing this a long time ago and it worked fine then.**

### ✔️ Expected result
CKEditor 5 retains the `type` attribute.

### ❌ Actual result
It does not.

## 📃 Other details

* Browser: N/A — reproduced in Chrome and Firefox.
* OS: N/A — reproduced in macOS Big Sur
* First affected CKEditor version: `v33.0.0`
* Installed CKEditor plugins: see above.

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by arkflpc**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Feature (list): Adds support for `type` attribute of `<ul>` and `<ol>` elements in addition to `list-style-type` style. Closes #11615.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

File: packages/ckeditor5-list/src/listproperties.js [hunk 1]
```diff
@@ -68,7 +68,7 @@ export default class ListProperties extends Plugin {
  * When set, the list style feature will be enabled. It allows changing the `list-style-type` HTML attribute of the lists.
```
Comment (oleq) [inline packages/ckeditor5-list/src/listproperties.js:68]: It's worth mentioning here that it's controlling either `list-style-type` style or the `type` attribute depending on the value.

File: packages/ckeditor5-list/src/listproperties/listpropertiesui.js [hunk 2]
```diff
@@ -277,7 +277,12 @@ function createListPropertiesView( {
 			listStyleCommand
 		} );
 
-		styleButtonViews = styleDefinitions.map( styleButtonCreator );
+		// The command can be ListStyleCommand or DocumentListStyleComman. Let's checks if it has expected function.
```
Comment (oleq) [inline packages/ckeditor5-list/src/listproperties/listpropertiesui.js:280]: `DocumentListStyleCommand`

File: packages/ckeditor5-list/ckeditor5-metadata.json [hunk 3]
```diff
@@ -87,7 +87,8 @@
 					],
 					"attributes": [
 						"start",
-						"reversed"
+						"reversed",
+						"type"
```
Comment (oleq) [inline packages/ckeditor5-list/ckeditor5-metadata.json:91]: Are you sure this is OK? Only the `DocumentListProperties` supports the `type` attribute. `ListProperties` is incapable of returning `type` in the data pipeline.

**Comment by wimleers** [comment]
YAY! This unblocks https://www.drupal.org/project/drupal/issues/3274635 — which means Drupal will be able to offer the superior list type UX that CKE5 supports, instead of the GHS-based work-around we were planning to ship with in https://www.drupal.org/project/drupal/issues/3274651 👍 🥳

**Pull request merged**